cli/glnexus_cli.cc

```diff
@@ -490,6 +490,212 @@ int main_dump(int argc, char *argv[]) {
+    }
+
+    cerr << "Loading discovered alleles "<< endl;
+    YAML::Node yaml = YAML::LoadFile(name);
+    if (!yaml) {
+        return GLnexus::Status::NotFound("bad discovered alleles file", name);
+    }
+    if (!yaml.IsSequence()) {
+        return GLnexus::Status::Invalid("not a sequence at top level");
+    }
+
+    valleles.clear();
+    for (YAML::const_iterator p = yaml.begin(); p != yaml.end(); ++p) {
+        GLnexus::discovered_alleles dsals;
+        s = discovered_alleles_of_yaml(*p, contigs, dsals);
+        if (s.bad()) return s;
+        valleles.push_back(dsals);
+    }
+
+    unsigned discovered_allele_count=0;
+    for (const auto& dsals : valleles) {
+        discovered_allele_count += dsals.size();
+    }
+    console->info() << "read " << discovered_allele_count << " discovered alleles";
+
+    return GLnexus::Status::OK();
+}
+
+GLnexus::Status parse_ranges_from_cmd_line(const string &bedfilename,
+                                           const string &range_txt,
+                                           const std::vector<std::pair<std::string,size_t> > &contigs,
+                                           vector<GLnexus::range> &ranges) {
+    if (bedfilename.empty()) {
+        // single range from the command line
+        GLnexus::range range(-1,-1,-1);
+        if (!parse_range(contigs, range_txt, range)) {
+            return GLnexus::Status::Invalid("range: ", range_txt);
+        }
+        ranges.push_back(range);
+    } else {
+        // read BED file
+        string rname, beg_txt, end_txt;
+        ifstream bedfile(bedfilename);
+        while (bedfile >> rname >> beg_txt >> end_txt) {
+            int rid = 0;
+            for(; rid<contigs.size(); rid++)
+                if (contigs[rid].first == rname)
+                    break;
+            if (rid == contigs.size()) {
+                return GLnexus::Status::Invalid("Unknown contig ", rname);
+            }
+            ranges.push_back(GLnexus::range(rid,
+                                            strtol(beg_txt.c_str(), nullptr, 10),
+                                            strtol(end_txt.c_str(), nullptr, 10)));
+        }
+        if (bedfile.bad() || !bedfile.eof()) {
+            return GLnexus::Status::IOError( "Error reading ", bedfilename);
+        }
+    }
+
+    sort(ranges.begin(), ranges.end());
+    for (auto& query : ranges) {
+        if (query.beg < 0 || query.end < 1 || query.end <= query.beg) {
+            return GLnexus::Status::Invalid("query range ", query.str(contigs));
+        }
+        if (query.end > contigs[query.rid].second) {
+            query.end = contigs[query.rid].second;
+            console->warn() << "Truncated query range at end of contig: " << query.str(contigs);
+        }
+    }
+
+    return GLnexus::Status::OK();
+}
+
+
+void help_discover(const char* prog) {
+    cerr << "usage: " << prog << " discover [options] /db/path " << endl
```

Comment:
The usage message should also show the "[chrom:1234-2345]" optional positional argument